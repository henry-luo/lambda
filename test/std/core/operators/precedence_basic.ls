// Test: Operator Precedence Basic
// Category: core/operators
// Type: positive
// Expected: 14

// Test multiplication before addition (2 + 3 * 4 = 2 + 12 = 14)
2 + 3 * 4
