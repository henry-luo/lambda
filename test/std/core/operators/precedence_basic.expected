09:15:47 [NOTICE] Start transpiling test/std/core/operators/precedence_basic.ls...
[script:int]
  [binary expr +.3:int]
    [primary expr:int,const:1]
      [const: int]
    [binary expr *.5:int]
      [primary expr:int,const:1]
        [const: int]
      [primary expr:int,const:1]
        [const: int]
09:15:47 [NOTICE] compiling C code in 'test/std/core/operators/precedence_basic.ls' to MIR
C2MIR init end           -- 72 usec
test/std/core/operators/precedence_basic.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2398, parse tokens -- 1441
  C2MIR preprocessor end    -- 501 usec
  C2MIR parser end          -- 628 usec
  C2MIR context checker end -- 712 usec
  C2MIR generator end       -- 755 usec
C2MIR compiler end                -- 789 usec
09:15:47 [NOTICE] Generating native code...
##### Script 'test/std/core/operators/precedence_basic.ls' executed: #####
14
